### PR TITLE
Add filter(f, ::NamedTuple)

### LINKED ISSUE #43473: `filter` a `NamedTuple`
`filter` currently does not have a method for `NamedTuple`. Given such a method does exist for dictionaries, maybe it makes sense to extend it in the same way?

### PR DESCRIPTION
**Pull request by aplavin**
https://github.com/JuliaLang/julia/pull/32968 added filter(Tuple), here I also extend it to NamedTuple
Closes https://github.com/JuliaLang/julia/issues/43473.

Typestable when possible, eg
```julia
julia> @code_warntype filter(x -> x isa Number, (a=1, b="2", c=3.))
<...>
Body::NamedTuple{(:a, :c), Tuple{Int64, Float64}}
<...>
```

### CODE REVIEW

File: base/namedtuple.jl [hunk 1]
```diff
@@ -269,6 +269,8 @@ function map(f, nt::NamedTuple{names}, nts::NamedTuple...) where names
     NamedTuple{names}(map(f, map(Tuple, (nt, nts...))...))
 end
 
+filter(f, xs::NamedTuple) = xs[filter(k -> f(xs[k]), keys(xs))]
```
Comment (akdienes) [inline base/namedtuple.jl:272]: I don't think `keys` guarantees a return order

**Comment by timholy** [comment]
The tests all fail 🙂

**Comment by LilithHafner** [comment]
Given that we have this for tuples, triage agrees that we should also have it for named tuples.

**Comment by LilithHafner** [comment]
Nothing on your end, sorry we abandoned this. LGTM

**Comment by LilithHafner** [comment]
Thanks!

**Pull request merged**